exonum-java-binding/testkit/src/test/java/com/exonum/binding/testkit/TestKitTest.java

```diff
@@ -426,6 +445,73 @@ void getAuditorEmulatedNode() {
     }
   }
 
+  @Test
+  void timeServiceWorksInTestKit() {
+    FakeTimeProvider timeProvider = FakeTimeProvider.create(TIME);
+    try (TestKit testKit = TestKit.builder(EmulatedNodeType.VALIDATOR)
+        .withService(TestServiceModule.class)
+        .withTimeService(timeProvider)
+        .build()) {
+      // Commit two blocks for time oracle to prepare consolidated time
+      testKit.createBlock();
+      testKit.createBlock();
+      testKit.withSnapshot((view) -> {
+        TimeSchema timeSchema = TimeSchema.newInstance(view);
+        Optional<ZonedDateTime> consolidatedTime = timeSchema.getTime().toOptional();
+        assertThat(consolidatedTime).contains(TIME);
+
+        // Check that validatorsTimes contains one exactly entry with TestKit emulated node's
+        // public key and time provider's time
+        checkValidatorsTimes(timeSchema, testKit);
+        return null;
+      });
+
+      // Update time in time provider
+      ZonedDateTime newTime = TIME.plusDays(1);
+      timeProvider.setTime(newTime);
+      testKit.createBlock();
+      testKit.createBlock();
+      testKit.withSnapshot((view) -> {
+        TimeSchema timeSchema = TimeSchema.newInstance(view);
+        Optional<ZonedDateTime> consolidatedTime = timeSchema.getTime().toOptional();
+        assertThat(consolidatedTime).contains(newTime);
+        return null;
+      });
+    }
+  }
+
+  @Test
+  void timeServiceWithManyValidatorsDoesNotReturnConsolidatedTime() {
```

Comment:
If it is not a bug, it is a very unfortunate peculiarity that, I think, needs to be documented briefly.

@MakarovS , @vitvakatu, Have you managed to identify the cause?

Reply:
Yes, it's because on 4 nodes or more this condition is true - https://github.com/exonum/exonum/blob/master/services/time/src/transactions.rs#L136

Reply:
I think it must be simply forbidden to create a network with four+ validators in `build` as it makes no sense with the present testkit.

Reply:
… *and* active time-service, obviously. That, however, will make it impossible to have time-service and test a service on 4+ nodes, but that "testing" won't correspond to a real-world situation on 4+ nodes anyway (non-working testkit) :thinking: